### PR TITLE
[MAINTENANCE] Run all python versions on scheduled run

### PR DESCRIPTION
**Pull request by Kilo59**
Enable comprehensive python version tests to run for scheduled runs.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

**Comment by ghost** [comment]
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=8463&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

####

[Review these changes using an interactive CodeSee Map](https://app.codesee.io/r/reviews?pr=8463&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
CodeSee Map legend
</details>

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -67,7 +67,7 @@ jobs:
 
   docs-snippets:
     name: docs-snippets
-    #if: github.event.pull_request.draft == false
+    if: github.event.pull_request.draft == false
```
Comment (Kilo59) [inline .github/workflows/ci.yml:70]: @Shinnnyshinshin 
Skipping this step for draft PRs

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -217,7 +217,15 @@ jobs:
...
           - postgresql
-        python-version: ["3.8"]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        exclude:
+          # TODO: would like to adopt `actionlint` pre-commit hook
+          # but false positive here and inability to do an inline ignore
+          # prevents this https://github.com/rhysd/actionlint/issues/237
+          - python-version: ${{ github.event_name != 'schedule' && '3.9' }}
+          - python-version: ${{ github.event_name != 'schedule' && '3.10' }}
+          - python-version: ${{ github.event_name != 'schedule' && '3.11' }}
```
Comment (Kilo59) [inline .github/workflows/ci.yml:224]: I tried multiple variations trying to get this to work in one line and none of them worked.
I've seen some other solutions on stack overflow but they seemed overly complicated and/or non-intuitive.

**Review by billdirks — APPROVED** [review]
Thanks, LGTM! We should think about whether we want to do this for other steps. We can chat about that tomorrow during standup.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Shinnnyshinshin) [reply .github/workflows/ci.yml:70]: Sounds good. Thank you @Kilo59

**Pull request merged**